docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderControlOptionRun.java

```diff
@@ -87,10 +89,16 @@ public void execute(AbstractBuild<?, ?> build) throws DockerException, IOExcepti
 
         LOGGER.info("Pulling image " + xImage);
 
-        InputStream result = client.pullImageCmd(xImage).exec();
+        PullImageResultCallback resultCallback = new PullImageResultCallback() {
+            public void onNext(PullResponseItem item) {
+                if (item.getStatus() != null && item.getProgress() == null) {
```

Comment:
All this control classes need to be updated and provide listener for printing output.